libsolidity/formal/SMTEncoder.cpp

```diff
@@ -923,9 +944,10 @@ void SMTEncoder::endVisit(IndexAccess const& _indexAccess)
 		return;
 	}
 
-	solAssert(array, "");
+	auto arrayVar = dynamic_pointer_cast<smt::SymbolicArrayVariable>(array);
```

Comment:
Hm... unrelated to this PR, but I wonder whether it makes sense to have a look at explicit ownership vs all the shared pointers in the smt code at some point...

Reply:
It does: https://github.com/ethereum/solidity/issues/7266